tests/phpunit/tests/block-bindings/postMetaSource.php

```diff
@@ -0,0 +1,263 @@
+<?php
+/**
+ * Tests for Block Bindings API "core/post-meta" source.
+ *
+ * @package WordPress
+ * @subpackage Blocks
+ * @since 6.5.0
+ *
+ * @group blocks
+ * @group block-bindings
+ */
+class Tests_Block_Bindings_Post_Meta_Source extends WP_UnitTestCase {
+	protected static $post;
+	protected static $wp_meta_keys_saved;
+
+	/**
+	 * Modify the post content.
+	 *
+	 * @param string $content The new content.
+	 */
+	private function getModifiedPostContent( $content ) {
+		self::$post->post_content = $content;
+		// Update the global $post variable to ensure all tests get the correct $post context.
+		$this->updateGlobalPost();
+		return apply_filters( 'the_content', self::$post->post_content );
+	}
+
+	/**
+	 * Update the global $post variable.
+	 */
+	private function updateGlobalPost() {
+		global $post;
+		$post = self::$post;
+	}
+
+	/**
+	 * Set up for every method.
+	 */
+	public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
+		self::$post               = $factory->post->create_and_get();
+		self::$wp_meta_keys_saved = isset( $GLOBALS['wp_meta_keys'] ) ? $GLOBALS['wp_meta_keys'] : array();
+	}
+
+	/**
+	 * Tear down for every method.
+	 */
```

Comment:
Incorrect. This is the tear down for the whole class.

```suggestion
	/**
	 * Tear down after class.
	 */
```